VulkanHppGenerator.cpp

```diff
@@ -11257,6 +11967,42 @@ bool VulkanHppGenerator::hasParentHandle( std::string const & handle, std::strin
   return false;
 }
 
+std::map<std::string, VulkanHppGenerator::DefineData> VulkanHppGenerator::calleeMacros( std::map<std::string, DefineData> defines ) const
+{
+  std::map<std::string, DefineData> calleeMacros{};
+
+  std::ranges::copy_if( defines,
+                        std::inserter( calleeMacros, calleeMacros.end() ),
+                        []( auto const & define )
+                        { return define.second.possibleCallee.empty() && !define.second.params.empty() && !define.second.possibleDefinition.empty(); } );
+
+  return calleeMacros;
+}
+
+std::map<std::string, VulkanHppGenerator::DefineData> VulkanHppGenerator::callerMacros( std::map<std::string, DefineData> defines ) const
+{
+  std::map<std::string, DefineData> callerMacros{};
+
+  std::ranges::copy_if( defines,
+                        std::inserter( callerMacros, callerMacros.end() ),
+                        []( auto const & define )
+                        { return !define.second.possibleCallee.empty() && !define.second.params.empty() && define.second.possibleDefinition.empty(); } );
+
+  return callerMacros;
+}
+
+std::map<std::string, VulkanHppGenerator::DefineData> VulkanHppGenerator::valueMacros( std::map<std::string, DefineData> defines ) const
+{
+  std::map<std::string, DefineData> valueMacros{};
+
+  std::ranges::copy_if( defines,
+                        std::inserter( valueMacros, valueMacros.end() ),
+                        []( auto const & define )
+                        { return define.second.possibleCallee.empty() && define.second.params.empty() && !define.second.possibleDefinition.empty(); } );
+
+  return valueMacros;
+}
```

Comment:
Those four functions (validMacros, calleeMacros, callerMacros, and valueMacros) effectively provide a pairwise disjoint partition of the m_defines. Which is not obvious from the code.
Maybe it's worth to better show that by a single partitioning function like:
```
VulkanHppGenerator::DefinesPartition VulkanHppGenerator::definesPartition( std::map<std::string, DefineData> const & defines ) const
{
  DefinesPartition partition;
  for ( auto const & define : defines )
  {
    if ( define.first.starts_with( "VK_" ) && ( define.first != "VK_DEFINE_HANDLE" ) )
    {
      if ( define.second.possibleCallee.empty() )
      {
        if ( define.second.possibleDefinition.empty() )
        {
#if !defined( NDEBUG )
          const std::set<std::string> ignoredDefines{ "VK_DEFINE_NON_DISPATCHABLE_HANDLE", "VK_NULL_HANDLE", "VK_USE_64_BIT_PTR_DEFINES" };
#endif
          assert( ignoredDefines.find( define.first ) != ignoredDefines.end() );
        }
        else
        {
          if ( define.second.params.empty() )
          {
            partition.values.insert( define );
          }
          else
          {
            partition.callees.insert( define );
          }
        }
      }
      else
      {
        assert( !define.second.params.empty() && define.second.possibleDefinition.empty() );
        partition.callers.insert( define );
      }
    }
  }
  return partition;
}
```